include/TLuaEngine.h

```diff
@@ -18,6 +18,9 @@
 #include <vector>
 
 #define SOL_ALL_SAFETIES_ON 1
+#define SOL_USER_C_ASSERT SOL_ON
+#define SOL_C_ASSERT(...) \
+    beammp_lua_errorf("SOL2 assertion failure: Assertion `{}` failed in {}:{}. This *should* be a fatal error, but BeamMP Server overrides it to not be fatal. This may cause the Lua Engine to crash, or cause other issues.", #__VA_ARGS__, __FILE__, __LINE__)
```

Comment:
To confirm, will this print out the file and line where the assert was called to assist the user in debugging & fixing / reporting?

Reply:
It prints out the file and line in the C++, not the Lua -- sadly an assert() means that there was a pretty much fatal error inside the lua engine, so yeah, the user should just report it.